ca_tools/scripts/control_node.py

```diff
@@ -0,0 +1,216 @@
+
+    def _stop(self):
+        aux = Twist()
+        self._my_pub.publish(aux)
+
+    def _move_forward(self):
+        aux = Twist()
+        aux.linear.x = LINEAR_VEL
+        self._my_pub.publish(aux)
+
+    def _forward(self):
+        if (self._reached_position()):
+            self._stop()
+            self.states = STATES.STOP_MOVING_FORWARD
+            self._goal_num += 1
+            self._goal_num %= 4
+        else:
+            self._set_goal_angle()
+            if (not self._reached_angle()):
+                self.states = STATES.STOP_MOVING_FORWARD
+                self._stop()
+            else:
+                self._move_forward()
+
+    def _diff_angle(self):
+        """Calculates the difference between the angle goal and current angle
+
+        Returns:
+            [float] -- [Difference of angles in radians, wrapped between -pi/2 and pi/2]
+        """
+        auxangle = (self._my_angle_goal-self._my_pose.theta)
+
+        while (abs(auxangle) > pi):
+            if auxangle < (-pi):
+                auxangle += (2*pi)
+            elif auxangle > pi:
+                auxangle -= (2*pi)
+        return auxangle
+
+    def _angle_is_big(self):
+        """Return if the angle difference between robot and goal is greater than a threshold
+        
+        Returns:
+            [boolean] -- [description]
+        """
+        return (abs(self._diff_angle()) > BIG_ANGLE_THRESHOLD)
+
+    def _stop_moving_forward(self):
+        self._stop()
+        self._set_goal_angle()
+        self.states = STATES.TURNING
+
+    def _set_goal_angle(self):
+        """Sets the goal angle with the current position and the goal position
+        """
+        self._my_angle_goal = math.atan2(
+            (self._my_goals[self._goal_num].y-self._my_pose.y), self._my_goals[self._goal_num].x-self._my_pose.x)
+
+    def _reached_angle(self):
+        """Returns true if reached angle
+        
+        Returns:
+            [bool] -- [description]
+        """
+        return (abs(self._diff_angle()) < ANGLE_TOLERANCE)
+
+    def _reached_position(self):
+        """Returns true if reached position
+        
+        Returns:
+            [bool] -- [description]
+        """
+        return (abs(self._diff_distance()) < DISTANCE_TOLERANCE)
+
+    def _reached_position(self):
+        """Checks if robot is at the goal, with some tolerance
+        Returns:
+        [bool] -- [true=->robot is at the goal]
+        """
+
+        return (abs(self._diff_distance()) < DISTANCE_TOLERANCE)
+
+    def _diff_distance(self):
+        """Returns the distance beteween the robot and the goal
+        """
+        return(math.hypot(self._my_goals[self._goal_num].x-self._my_pose.x, self._my_goals[self._goal_num].y-self._my_pose.y))
+
+
+if __name__ == '__main__':
+    node = CtrlNode()
+    # associates signal with handler
+    signal.signal(signal.SIGINT, signal_handler)
+    while not rospy.is_shutdown():
+        rospy.sleep(0.3)
```

Comment:
Why is this sleep here?

Reply:
Because `move()` was being called nonstop, so i could'nt stop the robot when it changes goal before entering to another call of `move()`, so it would do a little jump when reached a goal, i think there's a more elegant method, I don't know what it is though hehe